nanoFramework.Tools.DebugLibrary.Shared/DeviceConfiguration/NanoFrameworkDevices.cs

```diff
@@ -10,7 +13,13 @@ public class NanoFrameworkDevices : ObservableCollection<NanoDeviceBase>
 
         public static NanoFrameworkDevices Instance
         {
-            get { return _instance.Value; }
+            get
+            {
+                lock (_instance)

```

Comment:
Isn't this a bit resource intensive depending on what's done when the object is returned? That can fully block everything. So, is the returned value usage very small in terms of cycles or too long? In that case, you may have to find other ways.

Reply:
No, the lock is effective only for getting the instance of the list. The is done once when a Port*Manager is created, and once for every discovered device. If you are worried about performance: drop the Lazy, as it is hard to image a use of the library where the instance is not required.

Reply:
fine for me then!